hydrator-common/src/main/java/co/cask/hydrator/common/ReferenceBatchSink.java

```diff
@@ -38,5 +43,14 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
     // Verify that reference name meets dataset id constraints
     IdUtils.validateId(config.referenceName);
+    // create the external dataset this is done in the configure pipeline stage so that the external dataset with or
+    // without the schema is created when the pipeline is deployed itself.
+    Schema externalSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
```

Comment:
this logic isn't correct, as sinks may transform their input before writing. If the config has a schema, it should be used. If it does not have a schema, then no schema should be set.